### PR TITLE
update to julia v0.7 & v1.0

### PR DESCRIPTION
**Pull request by cecileane**
with documentation using Julia template

### CODE REVIEW

**Comment by cecileane** [comment]
I wonder why coverage has dropped so much. The test suite should note have changed!

**Comment by cecileane** [comment]
could `@test_logs` be interpreted as testing logging messages only, not other code that's run?

File: .travis.yml [hunk 1]
```diff
@@ -17,15 +21,24 @@ addons:
 
 after_success:
   # push coverage results to Coveralls
-  - julia -e 'cd(Pkg.dir("PhyloNetworks")); Pkg.add("Coverage"); using Coverage; Coveralls.submit(Coveralls.process_folder())'
+  - julia -e 'using Pkg; cd(Pkg.dir("PhyloNetworks")); Pkg.add("Coverage"); using Coverage; Coveralls.submit(Coveralls.process_folder())'
```
Comment (pbastide) [inline .travis.yml]: I don't know if that could help with the coverage issue, but it seems that there is a dedicated Travis option for using codecov and coveralls. Apparently, we could replace these `after_success` lines by (see https://docs.travis-ci.com/user/languages/julia/#coverage):
```yaml
codecov: true
coveralls: true
```
Maybe using the default could help catch all the tests ?

Comment (cecileane) [reply .travis.yml]: I noticed, but I have not seen this used. Actually, [NearestNeighborDescent](https://github.com/dillondaudert/NearestNeighborDescent.jl/blob/master/.travis.yml) uses both ways... Let’s give it a try. I’m not sure that these 2 lines should be part of the after_success phase (the Travis doc is not clear).

**Comment by cecileane** [comment]
The Travis configuration is simpler like you suggested (should be more robust), and the coverage diff is as before. The way coverage is calculated might have changed. Or I might have made a big mistake, like commented out a bunch of tests. But I don't think I did, and the coverage diff is fairly similar across all files.

**Comment by pbastide** [comment]
It looks like we are not the only one experiencing this issue, see: https://github.com/JuliaLang/julia/issues/28192. Julia people seem to be working on it. Maybe it will get fixed using the next version, or even nightly ? Anyway, it's probably not in our direct control, so I don't think that should stop us from carrying on.

File: src/descriptive.jl [hunk 2]
```diff
@@ -57,8 +56,8 @@ function fittedQuartetCF(d::DataCF, format=:wide::Symbol)
             tx3 = repeat([q.taxon[3] for q in d.quartet], inner=[3]),
             tx4 = repeat([q.taxon[4] for q in d.quartet], inner=[3]),
             quartet = repeat(["12_34","13_24","14_23"], outer=[nQ]),
-            obsCF = Array{Float64}(3*nQ),
-            expCF = Array{Float64}(3*nQ)  )
+            obsCF = Array{Float64}(undef, 3*nQ),
```
Comment (crsl4) [inline src/descriptive.jl]: Sorry, just out of curiosity, what does the "undef" refer to?

Comment (cecileane) [reply src/descriptive.jl]: `undef` means undefined: it's a function used to construct Arrays without initializing the values. We used to be able to allocated memory (without defining values --just taking bits as they came) like this:
`Array{Float64}(3)` for a vector of 3 values.
But in Julia v1.0, this syntax is not allowed. We have to be explicit about values not being initialized:
`Array{Float64}(undef, 3)`
There are alternative constructions, like
`Array{Union{Missing, Float64}}(missing, 3)` to initialized with missing values.

**Comment by pbastide** [comment]
Thanks Cécile for carrying this on ! :tada:

**Pull request merged**